pr_watcher_notifier/test_notification.py

```diff
@@ -0,0 +1,38 @@
+import json
+import textwrap
+from pathlib import Path
+
+import yaml
+
+from . import mail
+from .notification import make_email
+from .views import combine_data
+
+
+TEST_DATA = Path(__file__).parent.parent / "test_data"
+
+def test_sending(app, client):
+    with open(TEST_DATA / "pr_145_merged.json") as _:
+        data = json.load(_)
+    with open(TEST_DATA / "edx_config.yml") as _:
+        config = yaml.safe_load(_)
+    combine_data(data, config)
```

Comment:
@nedbat, this function call eventually ends up making an actual GitHub API call and by default that fails because the `GITHUB_ACCESS_TOKEN` is set to a dummy value as the unit tests (in which all external interactions are mocked) aren't expected to depend on any external services.

The easy fix will be to just update `test_settings.py` to read the GitHub access token from the environment (I'm not a big fan of using an actual API token). But the better one will be to mock the `get_pr()` function call to return just the values used in the data that an API call will actually return.

Reply:
Thanks, I had missed that this made a GitHub request.  I've mocked `get_pr`, and extended your original fake_pr code to take filenames.  This could mean we don't have to mock fnmatch any more, but I didn't go ahead and do that.